packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-action.ts

```diff
@@ -0,0 +1,477 @@
+  }
+
+  /**
+   * Render the actions in this chain
+   */
+  public renderActions(): CfnListener.ActionProperty[] {
+    return this.renumber([this.actionJson, ...this.next?.renderActions() ?? []]);
+  }
+
+  /**
+   * Called when the action is being used in a listener
+   */
+  public bindToListener(scope: Construct, listener: IApplicationListener, associatingConstruct?: IConstruct) {
+    // Empty on purpose
+    Array.isArray(scope);
+    Array.isArray(listener);
+    Array.isArray(associatingConstruct);
+  }
+
+  /**
+   * Renumber the "order" fields in the actions array.
+   *
+   * We don't number for 0 or 1 elements, but otherwise number them 1...#actions
+   * so ELB knows about the right order.
+   *
+   * Do this in `ApplicationListenerAction` instead of in `Listener` so that we give
+   * users the opportunity to override by subclassing and overriding `renderActions`.
+   */
+  protected renumber(actions: CfnListener.ActionProperty[]): CfnListener.ActionProperty[] {
+    if (actions.length < 2) { return actions; }
+
+    return actions.map((action, i) => ({ ...action, order: i + 1 }));
+  }
+}
+
+/**
+ * Options for `ApplicationListenerAction.forward()`
+ *
+ * @experimental
+ */
+export interface ForwardOptions {
+  /**
+   * The list of target groups to forward to
+   */
+  readonly targetGroups: IApplicationTargetGroup[];
+
+  /**
+   * For how long clients should be directed to the same target group
+   *
+   * Range between 1 second and 7 days.
+   *
+   * @default - No stickiness
+   */
+  readonly stickinessDuration?: Duration;
+}
+
+/**
+ * Options for `ApplicationListenerAction.weightedForward()`
+ *
+ * @experimental
+ */
+export interface WeightedForwardOptions {
+  /**
+   * The list of target groups to forward to
+   */
+  readonly targetGroups: WeightedTargetGroup[];
+
+  /**
+   * For how long clients should be directed to the same target group
+   *
+   * Range between 1 second and 7 days.
+   *
+   * @default - No stickiness
+   */
+  readonly stickinessDuration?: Duration;
+}
+
+/**
+ * A Target Group and weight combination
+ *
+ * @experimental
+ */
+export interface WeightedTargetGroup {
+  /**
+   * The target group
+   */
+  readonly targetGroup: IApplicationTargetGroup;
+
+  /**
+   * The target group's weight
+   *
+   * Range is [0..1000).
+   *
+   * @default 1
+   */
+  readonly weight?: number;
+}
+
+/**
+ * Options for `ApplicationListenerAction.fixedResponse()`
+ *
+ * @experimental
+ */
+export interface FixedResponseOptions {
+  /**
+   * Content Type of the response
+   *
+   * Valid Values: text/plain | text/css | text/html | application/javascript | application/json
+   *
+   * @default - Automatically determined
+   */
+  readonly contentType?: string;
+
+  /**
+   * The response body
+   *
+   * @default - No body
+   */
+  readonly messageBody?: string;
+
+  /**
+   * The HTTP response code
+   *
+   * Must be a 2xx, 4xx or 5xx response code.
+   */
+  readonly statusCode: number;
```

Comment:
I think this should be a positional argument (e.g. `fixedResponse(200)` and the rest as options)